scheds/rust/scx_mitosis/src/bpf/mitosis.bpf.c

```diff
@@ -438,27 +491,48 @@ static inline int update_task_cell(struct task_struct *p, struct task_ctx *tctx,
 		/*
 		 * Cgroup lookup failed - this can happen during scheduler load
 		 * for tasks that were forked before the scheduler was loaded,
-		 * whose cgroups went offline before scx_cgroup_init() ran.
-		 * Only fall back to root cgroup if the workaround is enabled
-		 * and the task is exiting.
+		 * whose cgroups went offline before scx_cgroup_init() ran, or
+		 * when CPU controller is not enabled (cgroup_init not called).
+		 *
+		 * If CPU controller is disabled, lazily initialize cgrp_ctx.
 		 */
-		if (exiting_task_workaround_enabled &&
-		    (p->flags & PF_EXITING)) {
-			struct cgroup *rootcg = READ_ONCE(root_cgrp);
-			if (!rootcg) {
-				scx_bpf_error(
-					"Unexpected uninitialized rootcg");
-				return -ENOENT;
-			}
-
-			cgc = lookup_cgrp_ctx(rootcg);
+		if (!READ_ONCE(cpu_ctrl_enabled) &&
+		    (cgc = bpf_cgrp_storage_get(
+			     &cgrp_ctxs, cg, 0,
+			     BPF_LOCAL_STORAGE_GET_F_CREATE))) {
```

Comment:
I really don't want cgrp_ctx creation happening on this path. I would have expected something similar to what's in layered that basically juts replicates cgroup_init off a tracepoint (https://github.com/sched-ext/scx/blob/main/scheds/rust/scx_layered/src/bpf/main.bpf.c#L812). Is there a reason that can't be made to work?

Reply:
I can give that a try, looks like it should work.